### PR TITLE
When branching a package also copy its url

### PR DESCRIPTION
**Pull request by vishaldeepak**
Hi,
In branch_package.rb when creating a new package also copy the url. This addresses the issue #11127 .

To see the changes 
- Goto any package example, http://localhost:3000/package/show/home:Admin/FirstPackage
- Click on Branch Package
- Hit branch
- The newly created branch should also have the URL

Added Rspec Test Case as necessary.

Here is a screenshot of how it looks:

Before
Screenshot 2021-07-08 at 12 38 43 AM

After:
Screenshot 2021-07-08 at 12 41 46 AM

### CODE REVIEW

File: src/api/spec/models/branch_package_spec.rb [hunk 1]
```diff
@@ -97,5 +97,14 @@
         end
       end
     end
+
+    context 'Newly created Package' do
+      it 'retains url' do
+        leap_project
+        branch_apache_package.branch
+        project = Package.where(name: 'apache2').where.not(id: apache.id).first
```
Comment (coolo) [inline src/api/spec/models/branch_package_spec.rb:105]: that `project` variable is actually a package. But that opens a side discussion: why not copy the url in the branch project also? It's just one bug report away

File: src/api/app/models/branch_package.rb [hunk 2]
```diff
@@ -273,12 +273,13 @@ def create_branch_project
 
       title = "Branch project for package #{params[:package]}"
       description = "This project was created for package #{params[:package]} via attribute #{@attribute}"
+      base_project_url = Project.find_by_name(params[:project]).try(:url)
```
Comment (vishaldeepak) [inline src/api/app/models/branch_package.rb:276]: is there a chance that params[:project] will not be received, also if there is a `params[:request]` should there be any changes?

File: src/api/spec/models/branch_package_spec.rb (hunk 1 above)
Comment (vishaldeepak) [reply src/api/spec/models/branch_package_spec.rb]: made changes as reqiured @coolo. One thing worth looking at - https://github.com/openSUSE/open-build-service/pull/11352/files#r666173752

File: src/api/app/models/branch_package.rb (hunk 2 above)
Comment (danidoni) [reply src/api/app/models/branch_package.rb:276]: Looking into the code base, right now `params[:project]` is always being passed but there is nothing in the code that prevents anyone from not passing it.

I don't know what to say about your second question... Maybe someone else can shed a bit of light?

Comment (vishaldeepak) [reply src/api/app/models/branch_package.rb:276]: I guess then its a misunderstanding from me, I think than its ready for a merge, unless there is any other comments

Comment (dmarcoux) [reply src/api/app/models/branch_package.rb:276]: The presence of `params[:request]` shouldn't affect the URL, so the changes are fine.

**Review by dmarcoux — CHANGES_REQUESTED** [review]
Thank you @vishaldeepak for submitting this PR. Could you rebase the PR on `master` to remove the unneeded merge commit `Merge pull request #11331 from vpereira/fix_11240`?

Everything else looks good.

**Comment by vishaldeepak** [comment]
Rebased with master. Let me know if there is some other priority issue that I could take up.

File: src/api/spec/models/branch_package_spec.rb [hunk 3]
```diff
@@ -97,5 +97,19 @@
         end
       end
     end
+
+    context 'retains url' do
+      it 'for a newly created package' do
+        branch_apache_package.branch
+        package = Package.where(name: 'apache2').where.not(id: apache.id).first
+        expect(package.url).to eq(apache.url)
```
Comment (dmarcoux) [inline src/api/spec/models/branch_package_spec.rb:105]: Somehow this spec is now failing? Were some of the changes lost in the rebase?

Comment (vishaldeepak) [reply src/api/spec/models/branch_package_spec.rb:105]: apologies, but it should be resolved now.

Comment (dmarcoux) [reply src/api/spec/models/branch_package_spec.rb:105]: No need to apologize, mistakes happen :slightly_smiling_face:

**Review by dmarcoux — APPROVED** [review]
Nice work @vishaldeepak!

**Pull request merged**